index.js

```diff
@@ -4,140 +4,219 @@ window.addEventListener("load", () => {
+    if (currentString != "") {
+      clearState = "c";
+      opButtons[5].textContent = clearState;
+    } else {
+      clearState = "ac";
+      opButtons[5].textContent = clearState;
+    }
+  }
+
+  function rounder(num) {
+    //This function accepts NUMBER data and returns STRING data.
+    if (Math.abs(num) > 999999999 || Math.abs(num) < 0.00000001) {
+      return num.toExponential(5);
+    } else {
+      let n = 10 - num.toString().split(".")[0].match(/\d/g).length;
+      //line 16 reads as follows: "Subtract the number of digits to the left of the decimal point from 10"
+      return parseFloat(num.toFixed(n)).toString();
+    }
+  }
+
+  function equation() {
+    if (currentOp == "add") {
+      currentString = firstNum + Number(currentString);
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+    } else if (currentOp == "subtract") {
+      currentString = firstNum - Number(currentString);
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+    } else if (currentOp == "multiply") {
+      currentString = firstNum * Number(currentString);
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+    } else if (currentOp == "divide") {
+      if (Number(currentString) == 0) {
+        numDisplay.textContent = "Error";
+        firstNum = "";
+        currentOp = "";
+        currentString = "";
+        errLock = true;
+      } else {
+        currentString = firstNum / Number(currentString);
+        numDisplay.textContent = rounder(Number(currentString));
+        firstNum = Number(currentString);
+        currentOp = "";
+        currentString = "";
+      }
+    } else {
+      return;
+    }
+  }
 
   for (let n = 0; n <= 9; n++) {
     numButtons[n].addEventListener("click", () => {
-      if (currentString.length <= 10) {
-        if (errLock == true) {
-          currentString = "";
-          errLock = false;
-          currentString = currentString + numButtons[n].textContent;
-          numDisplay.textContent = currentString;
-        } else {
-          currentString = currentString + numButtons[n].textContent;
-          numDisplay.textContent = currentString;
-        }
+      if (errLock == true) {
+        errLock = false;
+        currentString = numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
+      } else if (currentString == "") {
+        currentString = numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
+      } else if (currentString == "" || currentString == "0") {
+        currentString = numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
```

Comment:
These cases all seem to be pretty much the same. I think you could combine them:
```suggestion
      if (currentString == "" || currentString == "0") {
        errLock = false;
        currentString = numButtons[n].textContent;
        numDisplay.textContent = currentString;
        clearCheck();
```
I removed the check for errLock because currentString will be "" if errLock is true.

Reply:
Committed.